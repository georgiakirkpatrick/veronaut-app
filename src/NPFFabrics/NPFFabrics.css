.NPFFabrics .FormPromptWithSub {
    margin: 2vh 0 0;
}

.NPFFabrics .FormPopUp .FormButton {
    background-color: #F2E7F1;
}

.NPFFabrics .FormPopUp .FormPopUp__close-button {
    font-size: 12px;
}

.NPFFabrics .FormPopUp .FormPopUp__close-button {
    font-size: 12px;
}