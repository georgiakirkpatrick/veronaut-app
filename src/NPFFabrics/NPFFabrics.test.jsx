import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFFabrics from './NPFFabrics'

describe('NPFFabrics', () => {
    it ('renders without crashing', () => {
        render(<NPFFabrics />)
        screen.debug()
    })

    it ('render the UI as expected', () => {
        const tree = render(<NPFFabrics />)
        expect(tree).toMatchSnapshot()
    })
})