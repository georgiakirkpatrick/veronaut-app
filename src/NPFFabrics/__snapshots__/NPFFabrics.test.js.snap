// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFFabrics render the UI as expected 1`] = `
<div
  className="NPFFabrics"
  id="fabrics"
>
  <section
    className="FormPage"
  >
    <div
      className="empty"
    />
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Please enter the fabric details provided on the product webpage.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          If the product page does not provide the information detailed below, please select "not disclosed" from the dropdown.
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <h2>
          Fabric Dyeing, Printing, and Finishing
        </h2>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="dye-finish-country-id"
          >
            Dyeing and finishing country
          </label>
          
          <select
            name="dyeFinCountryId"
            onChange={[Function]}
            required={true}
            value=""
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="dye-fin-id"
          >
            Dyeing and finishing factory
          </label>
          
          <select
            name="dyeFinId"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              id={0}
              value={0}
            >
              Select a factory
            </option>
            <option
              id={1}
              value={1}
            >
              Not disclosed
            </option>
          </select>
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Factory not listed above?
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a factory
        </button>
        <div
          className="FormTextarea"
        >
          <label
            htmlFor="dye-fin-notes"
          >
            Whether or not the factory is listed, are there notes about the dyeing/printing/finishing mill?  If so, copy them and paste them here.
          </label>
          <textarea
            id="dye-fin-notes"
            name="dyeFinNotes"
            onChange={[Function]}
            required={true}
            rows="1"
            value=""
          />
        </div>
      </fieldset>
      <fieldset
        className="FormFieldset"
      >
        <h2>
          Knitting and Weaving
        </h2>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="wov-knit-country-id"
          >
            Knitting or weaving country
          </label>
          
          <select
            name="wovKnitCountryId"
            onChange={[Function]}
            required={true}
            value=""
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="wov-knit-id"
          >
            Knitting/weaving fabric mill
          </label>
          
          <select
            name="wovKnitId"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              id={0}
              value={0}
            >
              Select a fabric mill
            </option>
            <option
              id={1}
              value={1}
            >
              Not disclosed
            </option>
          </select>
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Factory not listed above?
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a fabric mill
        </button>
        <div
          className="FormTextarea"
        >
          <label
            htmlFor="wov-knit-notes"
          >
            Whether or not the fabric mill is listed, are there notes about where the fabric or material was made?  If so, copy them and paste them here.
          </label>
          <textarea
            id="wov-knit-notes"
            name="wovKnitNotes"
            onChange={[Function]}
            required={true}
            rows="1"
            value=""
          />
        </div>
      </fieldset>
      <fieldset
        className="FormFieldset"
      >
        <h2>
          Does the primary fabric have any of the following certifications?
        </h2>
        <div
          className="FormCheckboxSection"
        >
          <div
            className="empty"
          />
          <div
            className="FormCheckboxSection__options-container"
          />
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Certification not listed above?
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a certification
        </button>
      </fieldset>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Fibers
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          Please enter the fabric details provided on the product webpage.  If the product page does not provide the information detailed below, please select "not disclosed" from the dropdown.
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <button
          className="NewProductForm__remove"
          onClick={[Function]}
          type="button"
        >
          REMOVE
        </button>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="fiber-type-id-0"
          >
            Fiber or material
          </label>
          
          <select
            name="fiberTypeId"
            onChange={[Function]}
            required={true}
            value={0}
          >
            <option
              id={0}
              value={0}
            >
              Select a fiber or material
            </option>
            <option
              id={1}
              value={1}
            >
              Not disclosed
            </option>
          </select>
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Fiber or material not listed above?
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a fiber or material type
        </button>
        <section
          className="FormPopUp"
          id="fabric1new-fiber"
        >
          <button
            className="FormPopUp__close-button"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-xmark fa-lg "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
          <header>
            <h1>
              New Fiber
            </h1>
          </header>
          <div
            className="FormTextInput"
          >
            <label
              htmlFor="fabric1new-fiber-name"
            >
              Fiber name
            </label>
            <input
              id="fabric1new-fiber-name"
              name="name"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Submit fiber
          </button>
        </section>
        <div
          className="FormNumber"
        >
          <label
            htmlFor="fiber-type-percentage-0"
          >
            Percentage of fabric content, if available
          </label>
          <input
            id="fiber-type-percentage-0"
            name="percentage"
            onChange={[Function]}
            type="number"
            value=""
          />
           %
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="fiber-origin0"
          >
            Fiber origin
          </label>
          
          <select
            name="originId"
            onChange={[Function]}
            required={true}
            value={0}
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="producer-id0"
          >
            Fiber or material producer
          </label>
          
          <select
            name="producerId"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              id={0}
              value={0}
            >
              Select a producer
            </option>
            <option
              id={1}
              value={1}
            >
              Not disclosed
            </option>
          </select>
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Producer not listed above?
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a producer
        </button>
        <section
          className="FormPopUp"
          id="fabric1NewFibProducer"
        >
          <button
            className="FormPopUp__close-button"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-xmark fa-lg "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
          <header>
            <h1>
              New Producer
            </h1>
          </header>
          <div
            className="FormTextInput"
          >
            <label
              htmlFor="1NewProducerName"
            >
              Factory name
            </label>
            <input
              id="1NewProducerName"
              name="name"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="1NewProducerLocation"
            >
              Location
            </label>
            
            <select
              name="countryId"
              onChange={[Function]}
              required={true}
              value={0}
            />
          </div>
          <div
            className="FormUrlInput"
          >
            <label
              htmlFor="1NewProducerWebsite"
            >
              Website
            </label>
            <input
              id="1NewProducerWebsite"
              name="website"
              onChange={[Function]}
              required={true}
              type="url"
              value=""
            />
          </div>
          <div
            className="FormTextarea"
          >
            <label
              htmlFor="1NewProducerNotes"
            >
              Notes
            </label>
            <textarea
              id="1NewProducerNotes"
              name="notes"
              onChange={[Function]}
              required={true}
              rows="1"
              value=""
            />
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Submit Factory
          </button>
        </section>
        <div
          className="FormTextarea"
        >
          <label
            htmlFor="producer-notes0"
          >
            Whether or not the producer is listed, are there notes about the fiber or material producer?  If so, copy them and paste them here.
          </label>
          <textarea
            id="producer-notes0"
            name="producerNotes"
            onChange={[Function]}
            required={true}
            rows="1"
            value=""
          />
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Does the fiber have any of the following certifications?
          </p>
        </div>
        <div
          className="FormCheckboxSection"
        >
          <div
            className="empty"
          />
          <div
            className="FormCheckboxSection__options-container"
          />
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Certification not listed above?
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a certification
        </button>
      </fieldset>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          Does this fabric have more than one fiber?
        </p>
      </div>
      <button
        className="FormButton"
        onClick={[Function]}
        type="button"
      >
        Add a fiber
      </button>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
  <section
    className="FormPopUp"
    id="1NewFact"
  >
    <button
      className="FormPopUp__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-lg "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header>
      <h1>
        New Factory
      </h1>
    </header>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="1NewFactName"
      >
        Factory name
      </label>
      <input
        id="1NewFactName"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="FormDropdown"
    >
      <label
        htmlFor="1NewFactLocation"
      >
        Location
      </label>
      
      <select
        name="countryId"
        onChange={[Function]}
        required={true}
        value=""
      />
    </div>
    <div
      className="FormUrlInput"
    >
      <label
        htmlFor="1NewFactWebsite"
      >
        Website
      </label>
      <input
        id="1NewFactWebsite"
        name="website"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
    </div>
    <div
      className="FormTextarea"
    >
      <label
        htmlFor="1NewFactNotes"
      >
        Notes
      </label>
      <textarea
        id="1NewFactNotes"
        name="notes"
        onChange={[Function]}
        required={true}
        rows="1"
        value={1}
      />
    </div>
    <button
      className="FormButton"
      onClick={[Function]}
      type="button"
    >
      Submit Factory
    </button>
  </section>
  <section
    className="FormPopUp"
    id="1NewMill"
  >
    <button
      className="FormPopUp__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-lg "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header>
      <h1>
        New Fabric Mill
      </h1>
    </header>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="1-new-mill-name"
      >
        Fabric mill name
      </label>
      <input
        id="1-new-mill-name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="FormDropdown"
    >
      <label
        htmlFor="1NewMillLocation"
      >
        Location
      </label>
      
      <select
        name="countryId"
        onChange={[Function]}
        required={true}
        value=""
      />
    </div>
    <div
      className="FormUrlInput"
    >
      <label
        htmlFor="1NewMillWebsite"
      >
        Website
      </label>
      <input
        id="1NewMillWebsite"
        name="website"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
    </div>
    <div
      className="FormTextarea"
    >
      <label
        htmlFor="1NewMillNotes"
      >
        Notes
      </label>
      <textarea
        id="1NewMillNotes"
        name="notes"
        onChange={[Function]}
        required={true}
        rows="1"
        value=""
      />
    </div>
    <button
      className="FormButton"
      onClick={[Function]}
      type="button"
    >
      Submit fabric mill
    </button>
  </section>
  <section
    className="FormPopUp"
    id="1NewCert"
  >
    <button
      className="FormPopUp__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-lg "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header>
      <h1>
        New Certification
      </h1>
    </header>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="1-new-cert-name"
      >
        Certification name
      </label>
      <input
        id="1-new-cert-name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="FormUrlInput"
    >
      <label
        htmlFor="1-new-cert-website"
      >
        Website
      </label>
      <input
        id="1-new-cert-website"
        name="website"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
    </div>
    <button
      className="FormButton"
      onClick={[Function]}
      type="button"
    >
      Submit certification
    </button>
  </section>
</div>
`;
